documentation/modules/proc-accessing-kafka-using-routes.adoc

```diff
@@ -40,6 +40,8 @@ spec:
   zookeeper:
     # ...
 ----
++
+WARNING: Make sure that the name of the listener does not exceed 63 characters.
```

Comment:
Actually, the name of the listener is only up to 11 characters. The 63 characters is limit for the route name which includes namespace, name of the cluster, name of the listener etc. So it is a bit more complicated. I would maybe suggest more generic notice:

```suggestion
WARNING: The overall length of the OpenShift Route address might have a limited length. The Route address is based on the name of the Kafka cluster, name of the listener, and the name of the namespace they are created in. Please keep this in mind when naming the namespaces, Kafka clusters and listeners.
```

Reply:
This is pretty complex. An example might help.

Reply:
I've put the metadata in the example and updated the warning with example.
@scholzj - I think it's useful to include the limit?

> For example:
> 
> ```
> apiVersion: {KafkaApiVersion}
> kind: Kafka
> metadata:
>   labels:
>     app: my-cluster
>   name: my-cluster
>   namespace: myproject
> spec:
>   kafka:
>     # ...
>     listeners:
>       - name: external
>         port: 9094
>         type: route
>         tls: true
>         # ...
>     # ...
>   zookeeper:
>     # ...
>```
>
> WARNING: An OpenShift Route address comprises the name of the Kafka cluster, the name of the listener, and the name of the namespace it is created in. For example, `my-cluster-external-myproject`. Be careful with naming as the whole length of the address must not exceed 63 characters.

Reply:
Which limit do you mean? The 11 characters? Or the 63 characters?

Reply:
Actually both might be an idea.:

WARNING: An OpenShift Route address comprises the name of the Kafka cluster, the name of the listener, and the name of the namespace it is created in. For example, `my-cluster-external-myproject`. Be careful with naming as the whole length of the address must not exceed 63 characters, with an 11-character limit for the listener name.

Reply:
Well, the 11 character limit is general. So I think if we want to mention it I would not cover it here but in some more general section (it is already in the API reference of course). I would maybe not put it here as I think it is too confusing if we don't want to go into all the details how the routes are named.

Reply:
Thanks. Let's remove the 11 character bit.

Then we can have the note with or without the 63 char ref:

> WARNING: An OpenShift Route address comprises the name of the Kafka cluster, the name of the listener, and the name of the namespace it is created in. For example, `my-cluster-external-myproject`. Be careful that the whole length of the address does not exceed a maximum limit of 63 characters.

Or

> WARNING: An OpenShift Route address comprises the name of the Kafka cluster, the name of the listener, and the name of the namespace it is created in. For example, `my-cluster-external-myproject`. Be careful that the whole length of the address does not exceed a maximum limit.

Reply:
I'm not sure how often these two change ... but we can keep the 63 characters there I guess. Please also look at the other comment I left below.

Reply:
Thanks. will keep _63 characters_. I guess it's good practice to keep under that anyway